### PR TITLE
Fix multi dimensional array validation issue

### PR DESCRIPTION
**Pull request by fullkomnun**
## Description

Currently for a multi-dimensional array (bounded or unbounded) that follows a sibling field (is not the first field defined in its level), extrwcted JSON scheme is missing some field.

Sample ABI

```json
[
   {
      "name":"x1",
      "type":"uint256[][]"
   },
   {
      "name":"x2",
      "type":"uint256[][]"
   },
   {
      "name":"x3",
      "type":"uint256"
   }
]
```

Expected JSON Scheme 

```json
  {
   "type":"array",
   "items":[
      {
         "type":"array",
         "items":[
            {
               "type":"array",
               "$id":"x1",
               "items":{
                  "format":"uint",
                  "required":true
               }
            }
         ]
      },
      {
         "type":"array",
         "items":[
            {
               "type":"array",
               "$id":"x2",
               "items":{
                  "format":"uint",
                  "required":true
               }
            }
         ]
      },
      {
         "$id":"x3",
         "format":"uint256",
         "required":true
      }
   ],
   "maxItems":3,
   "minItems":3
}
```

Actual JSON Scheme 

```json
{
   "type":"array",
   "items":{
      "type":"array",
      "items":[
         {
            "type":"array",
            "$id":"x2",
            "items":{
               "format":"uint",
               "required":true
            }
         },
         {
            "$id":"x3",
            "format":"uint256",
            "required":true
         }
      ]
   },
   "maxItems":3,
   "minItems":3
}
```

This is due to mishandling of multi-dimensional arrays when ABI is converted to JSON scheme leading to loss of parts of the scheme (in this case 'x1' input parameter from the ABI is discarded). 
Fixed that and added a unit test payload to verify.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run lint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Review by avkos — APPROVED** [review]

File: packages/web3-validator/src/utils.ts [hunk 1]
```diff
@@ -187,7 +187,16 @@ export const abiSchemaToJsonSchema = (
...
-			lastSchema.items = childSchema;
+			// lastSchema.items is a Schema, concat with 'childSchema'
+			if (!Array.isArray(lastSchema.items)) {
+				lastSchema.items = [lastSchema.items as JsonSchema, childSchema];
+			} // lastSchema.items is an empty Scheme array, set it to 'childSchema'
+			else if (lastSchema.items.length === 0) {
+				lastSchema.items = childSchema;
+			} // lastSchema.items is a non-empty Scheme array, append 'childSchema'
+			else {
+				lastSchema.items.push(childSchema);
```
Comment (jdevcs) [inline packages/web3-validator/src/utils.ts:198]: need a unit test hitting this `lastSchema.items.push(childSchema);`

**Review by jdevcs — COMMENTED** [review]
Thanks for your contribution.

**Comment by jdevcs** [comment]
@fullkomnun also now there is need to fix Conflicting file. thanks.

**Review by jdevcs — APPROVED** [review]

File: packages/web3-validator/src/utils.ts (hunk 1 above)
Comment (fullkomnun) [reply packages/web3-validator/src/utils.ts:198]: Will do that in a follow-up PR (with another fix that covers another use-case for the implementation)

Comment (jdevcs) [reply packages/web3-validator/src/utils.ts:198]: Thanks @fullkomnun

**Pull request merged**